projects/sel4/porting.md

```diff
@@ -7,79 +7,159 @@ SPDX-FileCopyrightText: 2020 seL4 Project a Series of LF Projects, LLC.
 This is also useful for debugging, as you can use the FDT file system to dump addresses from a running system.
 
 Once you have found the correct address to load, you can use the command `bootcmd="tftpboot 0x20000000 sel4_image && go 0x20000000"`
-(using 0x20000000 as an example) followed by `saveenv` to save the starting address onto the boards flash memory.
-You will also need to set the 'ipaddr' and 'serverip' details can be found in the link above.
+(using `0x20000000` as an example) followed by the `saveenv` command to save the starting address onto the boards flash memory.
+
+You will also need to set the `ipaddr` and `serverip` environment variables, details can be found in the link above.
 
 ## seL4
 
-In order to have a successful port of seL4 to your target ARM platform you will need to go through the following steps.
+In order to have a successful port of seL4 to your target ARM or RISC-V platform you will need to go through the following steps.
+
+### Device Tree
+
+Device Trees are used by ARM and RISC-V platforms to describe the hardware platform.
+seL4 makes use of Device Trees at build time, which is why acquiring a Device Tree Source (DTS)
+for your platform is one of the first steps.
+
+seL4 relies on the DTS for the location and size of main memory as well as information
+about certain devices used by the kernel such as the timer. In debug mode, seL4 also
+makes use of the default serial device for debug output.
 
-### DTS
+Typically the Device Tree Source will be included in the
+[Linux kernel source](https://github.com/torvalds/linux), from there you can decompile
+the device tree blob (DTB) from building the Linux kernel to get the Device Tree that
+Linux uses.
 
-Look for DTS and drivers inside the [Linux repository](https://github.com/torvalds/linux), normally they will be supported. If they are supported clone Linux and modify the `update-dts.sh` to add your target. Run the script located in `kernel/tools/dts/update-dts.sh` at the cloned Linux repository. This will create a "platform.dts" file from Linux inside the dts folder in the kernel. You may need to add device nodes depending on what Linux uses and what is required by seL4. For example, in the port of the Rockpro64 a memory node and an extra timer node were needed so these had to be defined manually.
+You can compile the Linux Device Tree Sources into a final Device Tree Blob with:
+```sh
+make ARCH=<arm/arm64/riscv> CROSS_COMPILE=<TOOLCHAIN> defconfig
+make ARCH=<arm/arm64/riscv> CROSS_COMPILE=<TOOLCHAIN> dtbs
+```
+
+The DTBs can be found in `arch/<ARCH>/boots/dts/`.
+
+The following command allows you to decompile the DTB:
+```sh
+dtc -I dtb -O dts /path/to/dtb -o linux.dts
+```
+
+It should be noted that you may need to add device nodes depending on what Linux uses
+and what is required by seL4. For example, in the port of the ROCKPro64 a memory node
+and an extra timer node were needed so these had to be defined manually.
```

Comment:
Would it make sense to include a link to a diff or something similar for the ROCKPro64 change? Otherwise I'm not sure the specificity of the sentence adds much and might be more confusing than helping (i.e. people might wonder why ROCKPro64 is mentioned, whether it is something special etc). It's clear that it is an example, but it's not clear what is to be learned from the example.

Reply:
The other way of handling this would be to say somewhere at the beginning that we're using the ROCKPro64 port as a running example to illustrate the process.

Reply:
I have tried to make this section clearer now, let me now what you think.

Reply:
Yup, this is nice now!